src/napari_spatialdata/_viewer.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from anndata import AnnData
+from loguru import logger
+from multiscale_spatial_image import MultiscaleSpatialImage
+from napari import Viewer
+from napari.layers import Labels, Points, Shapes
+from napari.utils.notifications import show_info
+
+from napari_spatialdata.utils._utils import _get_transform, _swap_coordinates, _transform_to_rgb
+
+if TYPE_CHECKING:
+    from napari.layers import Layer
+    from spatialdata import SpatialData
+
+
+class SpatialDataViewer:
+    def __init__(self, viewer: Viewer) -> None:
+        self.viewer = viewer
+        self.viewer.bind_key("Shift-L", self._inherit_metadata)
+
+    def _inherit_metadata(self, viewer: Viewer) -> None:
+        layers = list(viewer.layers.selection)
+        self.inherit_metadata(layers)
+
+    def inherit_metadata(self, layers: list[Layer]) -> None:
+        """
+        Inherit metadata from active layer.
+
+        A new layer that is added will inherit from the layer that is active when its added, ensuring proper association
+        with a spatialdata object and coordinate space.
+
+        Paramters
+        ---------
+        layers: list[Layer]
+            A list of napari layers of which only 1 should have a spatialdata object from which the other layers inherit
+            metadata.
+        """
+        # Layer.metadata.get would yield a default value which is not what we want.
+        sdatas = [layer.metadata["sdata"] for layer in layers if "sdata" in layer.metadata]
+
+        sdata_ids = {id(sdata) for sdata in sdatas}
```

Comment:
I don't think we should use `id()` (see post below). If I understood correctly, you switched to `id()` because there were some issues with the counting of unique sdata objects. If I recall correctly, your previous sdata count was potentially off because you were doing something like `sum(sdata_a is sdata_b)`, which would count the number of times `sdata_a` is repeated, not the number of unique sdatas there are. 

https://stackoverflow.com/questions/52268343/id-vs-is-operator-is-it-safe-to-compare-ids-does-the-same-id-mean-the

Reply:
resolved with [e4761cf](https://github.com/scverse/napari-spatialdata/pull/120/commits/e4761cf5263b2b6191c0cbc668b0447a8ec1ba8c)